helm/resource_release.go

```diff
@@ -136,6 +136,14 @@ func resourceRelease() *schema.Resource {
 							Type:     schema.TypeString,
 							Required: true,
 						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "auto",
```

Comment:
When upgrading this causes a diff even though nothing has actually changed, which is confusing. I propose removing the default, as your `getValue()` func considers `"auto"` and `""` equivalent.